src/document.cpp

```diff
@@ -682,8 +682,13 @@ void REHex::Document::_save_metadata(const std::string &filename)
 {
 	/* TODO: Atomically replace file. */
 	
-	json_t *meta = _dump_metadata();
-	int res = json_dump_file(meta, filename.c_str(), JSON_INDENT(2));
+	bool has_data = false;
+	json_t *meta = _dump_metadata(has_data);
+	int res = 0;
+	if (has_data)
```

Comment:
Should have an else case here which unlinks the file - else it becomes impossible to delete the last comment.

Reply:
Added unlink using wxWidgets functions, since those are used also to check for the file initially.